### ISSUE TITLE
GCC 6.3.1 generates too large binaries for some targets

### ISSUE THREAD

**Issue by msplr** [cvra/can-bootloader#76]
when building for motor-board:
```
arm-none-eabi/bin/ld: bootloader.elf section `.rodata' will not fit in region `FLASH_TEXT'
arm-none-eabi/bin/ld: region `FLASH_TEXT' overflowed by 76 bytes
```

**Comment by antoinealb** [comment]
Interesting. Do you still have your previous GCC install to detect how much room we had before?

**Comment by msplr** [comment]
With `gcc-arm-none-eabi-4_9-2014q4` I get 9996 bytes, which leaves 244 bytes room.

**Comment by msplr** [comment]
Enabling link time optimization also fixes the issue (using gcc-arm-none-eabi-6-2017-q2).
You can do this by adding `-flto` to CFLAGS and LFLAGS in the Makefile.
I should fix this in a PR sometime...

**Comment by antoinealb** [comment]
Is `-flto` stable yet ? Last time I checked it was full of weird bugs.

Anyway #78 should be a less risky fix IMHO.

**Comment by SyrianSpock** [comment]
Solved by #78
